src/foundry/foundry.js/applications/formApplication.d.ts

```diff
@@ -128,7 +128,7 @@ declare abstract class FormApplication<
    * @returns The prepared update data
    */
   // TODO: Maybe we can calculate how the flattened `updateData` looks like, then it would be Partial<Record<string, unknown>> & Flattened<T>
-  protected _getSubmitData(updateData?: object): Partial<Record<string, unknown>>;
+  protected _getSubmitData(updateData?: object | null): Partial<Record<string, unknown>>;
```

Comment:
Where is this called with `null`?

Reply:
In `FormApplication#_onSubmit`, the default value for `updateData` is `null`, so `_getSubmitData` may subsequently be called with `null`.